test/phoenix_live_view/async_test.exs

```diff
@@ -0,0 +1,81 @@
+defmodule Phoenix.LiveView.AsyncTest do
+  # run with async: false to prevent other messages from being captured
+  use ExUnit.Case, async: false
+
+  import ExUnit.CaptureIO
+
+  describe "async operations" do
+    for fun <- [:assign_async, :start_async] do
+      test "warns when passing socket to #{fun} function" do
+        warnings =
+          capture_io(:stderr, fn ->
+            defmodule AssignAsyncSocket do
```

Comment:
Use different module names per test to avoid warnings?

Reply:
could do, although the warnings are captured by capture_io 😃

Reply:
oh, evil :D